Update RBAC best practices for workload creation (#36343)
kubernetes/website: content/en/docs/concepts/security/rbac-good-practices.md

### Workload creation

Permission to create workloads (either Pods, or
[workload resources](/docs/concepts/workloads/controllers/) that manage Pods) in a namespace

> implicitly grants access to many other resources in that namespace, such as Secrets, ConfigMaps, and
> PersistentVolumes, which can be mounted into a Pod. Additionally, since Pods can run as any

Reviewer: ```suggestion
implicitly grants access to many other resources in that namespace that can be mounted in Pods, such as Secrets, ConfigMaps, and
PersistentVolumes. Additionally, since Pods can run as any
```

> [ServiceAccount](/docs/reference/access-authn-authz/service-accounts-admin/), create workload
> permissions effectively grants the API access levels of any ServiceAccount in the namespace.

Reviewer: ```suggestion
[ServiceAccount](/docs/reference/access-authn-authz/service-accounts-admin/), granting permission to create workloads also implicitly grants the API access levels of any service account in that namespace.
```

You can use [Pod Security admission](/docs/concepts/security/pod-security-admission/)
or other (third party) mechanisms to implement that enforcement.

> For these reasons, it is considered best practice to generally grant permissions at the namespace
> level, and use caution when subdividing access within a namespace.

Reviewer: ```suggestion
For these reasons, as a best practice, grant permissions at the namespace
level and use caution when subdividing access within a namespace.
```
It's not clear what "subdividing access" means. What specifically should they be cautious about? What configuration could lead to issues?

Author: How's this?

> For these reasons, it is considered best practice to generally grant permissions at the namespace
level, and use caution when attempting to limit access to a subset of namespace resources.
